sdk/search/Azure.Search.Documents/tests/Batching/BatchingTests.cs

```diff
@@ -0,0 +1,985 @@
+            {
+                try
+                {
+                    SplitWhenRequested();
+                    return ProcessResponse(await base.IndexDocumentsAsync(batch, options, cancellationToken).ConfigureAwait(false));
+                }
+                finally
+                {
+                    RaiseNotification();
+                }
+            }
+        }
+
+        public BatchingSearchClient GetBatchingSearchClient(SearchResources resources)
+        {
+            return InstrumentClient(
+                new BatchingSearchClient(
+                    resources.Endpoint,
+                    resources.IndexName,
+                    new AzureKeyCredential(resources.PrimaryApiKey),
+                    GetSearchClientOptions()));
+        }
+        #endregion
+
+        #region Champion
+        [Test]
+        public async Task Champion_OneShotUpload()
+        {
+            await using SearchResources resources = await SearchResources.CreateWithEmptyIndexAsync<SimpleDocument>(this);
+            SearchClient client = resources.GetSearchClient();
+            SimpleDocument[] data = SimpleDocument.GetDocuments(50000);
+
+            // Wrap in a block so we DisposeAsync before getting the Count below
+            {
+                await using SearchIndexingBufferedSender<SimpleDocument> indexer =
+                    client.CreateIndexingBufferedSender<SimpleDocument>();
+                AssertNoFailures(indexer);
+                await indexer.UploadDocumentsAsync(data);
+            }
+
+            // Check that we have the correct number of documents
+            await WaitForDocumentCountAsync(resources.GetSearchClient(), data.Length);
+        }
+
+        [Test]
+        public async Task Champion_ContinueAddingWhileSending()
+        {
+            await using SearchResources resources = await SearchResources.CreateWithEmptyIndexAsync<SimpleDocument>(this);
+            SearchClient client = resources.GetSearchClient();
+            SimpleDocument[] data = SimpleDocument.GetDocuments(1000);
+
+            // Wrap in a block so we DisposeAsync before getting the Count below
+            {
+                await using SearchIndexingBufferedSender<SimpleDocument> indexer =
+                    client.CreateIndexingBufferedSender<SimpleDocument>();
+                AssertNoFailures(indexer);
+
+                // Trickle documents in
+                for (int i = 0; i < data.Length; i++)
+                {
+                    await indexer.UploadDocumentsAsync(new[] { data[i] });
+                    await DelayAsync(TimeSpan.FromMilliseconds(5));
+                }
+            }
+
+            // Check that we have the correct number of documents
+            await WaitForDocumentCountAsync(resources.GetSearchClient(), data.Length);
```

Comment:
Why not just use `client` from above? Seems odd to generate a new one. I see this below as well.

Reply:
The test infrastructure that listens for diagnostic scopes doesn't appear to be thread safe.  There can still be pending scopes on the client used by the publisher that cause these to flake when the timing is unlucky so I'm just creating fresh clients.